source/main.coffee

```diff
@@ -36,6 +36,14 @@ module.exports =
   compile: (src, options=defaultOptions) ->
     filename = options.filename or "unknown"
 
+    # TODO: This incorrectly shifts line numbers by 1
+    if filename.endsWith('.coffee') and not
+       /^(#![^\r\n]*(\r\n|\n|\r))?\s*['"]civet/.test src
+      src = """
+        "civet coffeeCompat"
```

Comment:
It may be possible to update the CivetPrologue rule to not be as strict about EOL and then to have this as
```
"civet coffeeCompat";#{src}
```